contracts/common/misc/Proxy.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity ^0.5.2;
+import { DelegateProxy } from "./DelegateProxy";
+
+
+contract Proxy is ProxyStorage, DelegateProxy {
+
+  event Upgrade(address indexed newContract, bytes initializedWith);
+  event OwnerUpdate(address _prevOwner, address _newOwner);
+
+  constructor() public {
+    proxyOwner = msg.sender;
+  }
+
+  //
+  // Dispatch fallback
+  //
+
+  function () public payable {
+    // require(currentContract != 0, "If app code has not been set yet, do not call");
+    // Todo: filter out some calls or handle in the end fallback
+    delegatedFwd(proxyTo, msg.data);
+  }
+
+  //
+  // Ownership
+  //
+
+  modifier onlyProxyOwner() {
+    require(msg.sender == proxyOwner, "Unauthorized user");
```

Comment:
I am currently using UNAUTHORIZED_USER typing for error msgs. Let's keep the same?